src/ray/gcs/gcs_server/gcs_actor_scheduler.cc

```diff
@@ -154,6 +154,36 @@ ActorID GcsActorScheduler::CancelOnWorker(const ClientID &node_id,
   return assigned_actor_id;
 }
 
+void GcsActorScheduler::ReleaseUnusedWorkers(
+    const std::unordered_map<ClientID, std::vector<WorkerID>> &node_to_workers,
+    const EmptyCallback &callback) {
+  RAY_CHECK(callback);
+  // If the node is dead, there is no need to send the request of release unused
+  // workers.
+  const auto &alive_nodes = gcs_node_manager_.GetAllAliveNodes();
+  // We need to send a request to each node, because it may be successful in the lease
+  // worker, but GCS server has not processed it before GCS server is restarted.
+  for (const auto &alive_node : alive_nodes) {
+    rpc::Address address;
+    address.set_raylet_id(alive_node.second->node_id());
+    address.set_ip_address(alive_node.second->node_manager_address());
+    address.set_port(alive_node.second->node_manager_port());
+    auto lease_client = GetOrConnectLeaseClient(address);
+    auto release_unused_workers_callback =
+        [callback](const Status &status, const rpc::ReleaseUnusedWorkersReply &reply) {
+          callback();
+        };
+    auto iter = node_to_workers.find(alive_node.first);
+    if (iter != node_to_workers.end()) {
+      RAY_CHECK_OK(lease_client->ReleaseUnusedWorkers(iter->second,
+                                                      release_unused_workers_callback));
+    } else {
+      RAY_CHECK_OK(
```

Comment:
Pls add some comment?

Reply:
fixed

Reply:
This comment isn't clear enough. Please take a look at my comment in the proto file.

Reply:
Also such a comment should be put in the beginning of this function.